src/cozmo/action.py

```diff
@@ -442,13 +448,22 @@ def _abort_action(self, action):
         # message back in the next engine tick, and can basically be considered
         # cancelled from now.
         action._set_aborting()
-        # move from in-progress to aborting dicts
-        self._aborting[action._action_id] = action
-        del self._in_progress[action._action_id]
 
-        msg = _clad_to_engine_iface.CancelActionByIdTag(idTag=action._action_id,
-                                                        robotID=self.robot.robot_id)
-        self.robot.conn.send_msg(msg)
+        if action._completed_event_pending:
+            # The action was marked as still running but the ActionDispatcher
+            # has already received a completion message (and removed it from
+            # _in_progress) - the action is just waiting to receive a
+            # robot_completed_action message that is still being dispatched
+            # via asyncio.ensure_future
+            logger.debug('Not sending abort for action=%s to engine as it just completed', action)
+        else:
+            # move from in-progress to aborting dicts
```

Comment:
I'm curious why this else case code is new and wasn't needed prior to this change (moving the _action_id from _aborting to _in_progress)?